### PR TITLE
Enable p2p public address definition

### PR DESCRIPTION
**Pull request by otherview**
### Why is this change needed?

- https://github.com/obscuronet/obscuro-internal/issues/640

### What changes were made as part of this PR:

- Splits p2pAddress into p2pBindAddress and p2pPublicAddress
- p2pBindAddress Where the server is bound (tipically 0.0.0.0:10000)
- p2pPublicAddress The address that is anounced to other peers (like obscuro-node-21.bla.com:1000)

### CODE REVIEW

File: go/host/host.go [hunk 1]
```diff
@@ -635,7 +635,7 @@ func (a *Node) processSharedSecretResponse(_ *ethadapter.L1RespondSecretTx) erro
 	var filteredHostAddresses []string //nolint:prealloc
 	for _, hostAddress := range hostAddresses {
 		// We exclude our own address.
-		if hostAddress == a.config.P2PAddress {
+		if hostAddress == a.config.P2PBindAddress {
```
Comment (tudor-malene) [inline go/host/host.go:638]: shouldn't this be the advertised address?

File: go/host/hostrunner/cli_flags.go [hunk 2]
```diff
@@ -32,8 +32,11 @@ const (
 	enclaveRPCTimeoutSecsName  = "enclaveRPCTimeoutSecs"
 	enclaveRPCTimeoutSecsUsage = "The timeout for host <-> enclave RPC communication"
 
-	p2pAddressName  = "p2pAddress"
-	p2pAddressUsage = "The P2P address for our node"
+	p2pBindAddressName  = "p2pBindAddress"
+	p2pBindAddressUsage = "The P2P address where the server is bound to. Defaults to 0.0.0.0:10000"
+
+	p2pPublicAddressName  = "p2pPublicAddress"
+	p2pPublicAddressUsage = "The P2P address where the server is bound to. Defaults to 127.0.0.1:10000"
```
Comment (tudor-malene) [inline go/host/hostrunner/cli_flags.go:39]: is this description correct?

File: go/host/host.go [hunk 3]
```diff
@@ -146,7 +146,7 @@ func (a *Node) Start() {
 			AggregatorID:  &a.ID,
 			Attestation:   encodedAttestation,
 			InitialSecret: a.EnclaveClient.GenerateSecret(),
-			HostAddress:   a.config.P2PAddress,
+			HostAddress:   a.config.P2PBindAddress,
```
Comment (tudor-malene) [inline go/host/host.go:149]: shouldn't the p2p public address be advertised here?

**Review by tudor-malene — COMMENTED** [review]
I'm a bit tired at this hour to wrap my head around the implications of this.
Can you have a look as well @BedrockSquirrel ?

File: go/host/host.go (hunk 3 above)
Comment (otherview) [reply go/host/host.go:149]: Yep, fixed :)

File: go/host/hostrunner/cli.go [hunk 4]
```diff
@@ -51,7 +52,8 @@ func ParseConfig() config.HostConfig {
 	clientRPCTimeoutSecs := flag.Uint64(clientRPCTimeoutSecsName, uint64(cfg.ClientRPCTimeout.Seconds()), clientRPCTimeoutSecsUsage)
 	enclaveRPCAddress := flag.String(enclaveRPCAddressName, cfg.EnclaveRPCAddress, enclaveRPCAddressUsage)
 	enclaveRPCTimeoutSecs := flag.Uint64(enclaveRPCTimeoutSecsName, uint64(cfg.EnclaveRPCTimeout.Seconds()), enclaveRPCTimeoutSecsUsage)
-	p2pAddress := flag.String(p2pAddressName, cfg.P2PAddress, p2pAddressUsage)
+	p2pBoundAddress := flag.String(p2pBindAddressName, cfg.P2PBindAddress, p2pBindAddressUsage)
```
Comment (BedrockSquirrel) [inline go/host/hostrunner/cli.go:55]: Minor but worth sticking to `bind` naming instead of `bound`? Easier to search if we're consistent

**Review by BedrockSquirrel — COMMENTED** [review]
Couple of minor things but generally change seems sensible to me

File: go/host/host.go (hunk 1 above)
Comment (otherview) [reply go/host/host.go]: Fixed ! :)

File: go/host/hostrunner/cli.go (hunk 4 above)
Comment (otherview) [reply go/host/hostrunner/cli.go:55]: Good point, fixed 👍

File: go/host/hostrunner/test.toml [hunk 5]
```diff
@@ -7,7 +7,8 @@ clientRPCHost = "127.0.0.1"
 clientRPCTimeoutSecs = 10
 enclaveRPCAddress = "127.0.0.1:11000"
 enclaveRPCTimeoutSecs = 10
-p2pAddress = "127.0.0.1:10000"
+p2pPublicAddress = "127.0.0.1:10000"
+p2pBindAddress = "0.0.0.1:10000"
```
[the remark this answers is not in the document]
Comment (BedrockSquirrel) [reply go/host/hostrunner/test.toml:11]: I mean I think you've got a typo in the bind address? `0.0.0.1` instead of `0.0.0.0`

**Review by tudor-malene — COMMENTED** [review]
lgtm (in principle), once all of Matt's comments were addressed.

File: go/host/hostrunner/test.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (otherview) [reply go/host/hostrunner/test.toml:11]: I did not see that typo in the BindAddress, thank you!

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**